File: day1/2.js
'use strict';
const fs = require('fs');

let fileName = process.argv[2] || 'input.txt';
fs.readFile(fileName, (err, data) => {
  console.log(getFirstBasementIndex(String(data)));
});

function getFirstBasementIndex(directions) {
  let currentFloor = 0;
  let firstBasement;

  directions.split('').some((floor, index) => {
    if (floor === '(') {
      currentFloor++;
    } else if (floor === ')') {
      currentFloor--;
    }

    if (currentFloor === -1) {
      firstBasement = index + 1;
      return true;
    }
  });

  return firstBasement;
}

File: day1/2.py
import fileinput

def getFirstBasementIndex(directionsFile):
    currentFloor = 0
    firstBasement = 0
    totalIndex = 0
    for line in directionsFile:
        for index, floor in enumerate(line):
            if floor == '(':
                currentFloor += 1
            elif floor == ')':
                currentFloor -= 1

            if currentFloor == -1:
                firstBasement = totalIndex + index + 1
                return firstBasement
        totalIndex += len(line) - 1

    return None

print getFirstBasementIndex(fileinput.input())
